geonode_mapstore_client/client/js/components/MediaViewer/Media.jsx

```diff
@@ -60,7 +48,28 @@ const mediaDefaultProps = {
     unsupported: {}
 };
 
-const Media = ({resource, ...props}) => {
+const Media = ({ resource, ...props }) => {
+    useEffect(() => {
+        const image = document.querySelector('.ms-media img');
+        // if an image is being viewed avoid stretching small images to fit viewer dimensions
+        if (image) {
+            let newimageWidth = 0;
+            let newimageHeight = 0;
+            let newimage = new Image();
+            newimage.src = resource.href;
+            newimage.onload = function() {
+                newimageWidth = this.naturalWidth;
+                newimageHeight = this.naturalHeight;
+                const container = document.getElementById('ms-container');
+                const containerStyles = window.getComputedStyle(container);
+                if (newimageWidth < parseInt(containerStyles.width, 10) && newimageHeight < parseInt(containerStyles.height, 10)) {
+                    return image.classList.add('natural-image');
+                }
+                return false;
+            };
+        }
+    }, []);
```

Comment:
This solution seems applied only on the mount this means it's not applied on resize.
Anyway I think we could use a pure css solution without using js.
I was testing some changes in the classes see this sample

```html
<!DOCTYPE html>
<html>
    <head>
        <style>
           .gn-media-viewer {
                height: calc(100% - 2rem);
                left: 0;
                margin: 1rem;
                position: absolute;
                top: 0;
                width: calc(100% - 2rem);
           }
           .gn-media-viewer .ms-media {
                position: relative;
                width: 100%;
                height: 100%;
            }
            .ms-media-image {
                display: flex;
                align-items: center;
                justify-content: center;
            }
            img {
                position: relative;
                max-width: 100%;
                max-height: 100%;
                object-fit: contain;
                cursor: pointer;
                width: auto;
                height: auto;
            }
        </style>
    </head>
    <body>
        <div class="gn-media-viewer">
            <div class="ms-media ms-media-image">
                <!-- large image -->
                <!-- img src="https://master.demo.geonode.org/documents/1673/link"/-->
                <!-- small image -->
                <img src="https://master.demo.geonode.org/documents/1577/link" />
            </div>
        </div>
    </body>
</html>
```
The main changes are related to apply only max-width and max-height to the img element and ensure that the ms-media-image is a flex that align the item on the center. Could you please test this solution?

Reply:
It works @allyoucanmap